### PR TITLE
Feature/camera logs

### PR DESCRIPTION
**Pull request by rogermiretin2**
-Camera-related logs management: captures errors from the built-in error listeners from zxing-scanner (scanError, scanFailure) as well as from qr http requests and from the fetching/parsing from the storage. Other errors captured by zxing-scanner are intercepted though console.error. Stack-traces are not stored due to length limitations when sending (see below). 
-Logs are stored through storage service (which uses Ionic storage --indexedDB). 
-Logs are shown in Settings>View Logs>Camera logs 
-Last logs are sent from Settings>Send logs through mailTo. Only 1400 characters, which is aprox. the minimum size allowed by local email services. 
-Feature-flag conditioned (logs-enabled) 
-Added tests with Jest. Since the migration isn't complete, **only those tests which are directly related to the camera logs feature are taken into account in the test execution and coverage**.

### CODE REVIEW

File: src/app/components/barcode-scanner/barcode-scanner.component.ts [hunk 1]
```diff
@@ -53,22 +61,57 @@ export class BarcodeScannerComponent implements OnInit {
...
   public scanSuccess$ = new BehaviorSubject<string>('');
-  public constructor(private cameraService: CameraService, private router: Router) {
-    this.router.events
+  public constructor(
+    private readonly cameraService: CameraService,
+    private readonly cameraLogsService: CameraLogsService, private router: Router) {
+      // Requires debounce since this type of error is emitted constantly
+      this.scanFailureSubject.pipe(
+        distinctUntilChanged((
+          a, b) =>
```
Comment (albertorubioin2) [inline src/app/components/barcode-scanner/barcode-scanner.component.ts:75]: en vez de a y b utilicemos una variable más descriptiva

File: src/app/components/barcode-scanner/barcode-scanner.component.ts [hunk 2]
```diff
@@ -53,22 +61,57 @@ export class BarcodeScannerComponent implements OnInit {
       distinctUntilChanged(),
       shareReplay(1)
     );
+    private readonly scanFailureSubject = new Subject<Error>();
+    private readonly scanFailureDebounceDelay = 3000;
+    private originalConsoleError: any;
```
Comment (albertorubioin2) [inline src/app/components/barcode-scanner/barcode-scanner.component.ts:66]: tipo?

File: src/app/pages/credentials/credentials.page.spec.ts [hunk 3]
```diff
@@ -65,57 +92,91 @@ describe('CredentialsPage', () => {
...
+    expect(component.ebsiFlag).toBe(false);
   });
 
-  it('should handle error gracefully if clipboard API fails', async () => {
-    spyOn(navigator.clipboard, 'writeText').and.rejectWith('Test error');
-    spyOn(console, 'error');
 
-    const didTextContent = 'DID: exampleDid';
-    document.body.innerHTML = `<div id="did-text">${didTextContent}</div>`;
+  // TODO
```
Comment (albertorubioin2) [inline src/app/pages/credentials/credentials.page.spec.ts:104]: eliminar cometarios

File: src/app/components/barcode-scanner/barcode-scanner.component.ts (hunk 2 above)
Comment (rogermiretin2) [reply src/app/components/barcode-scanner/barcode-scanner.component.ts:66]: hi poso private originalConsoleError: (...data: any[]) => void; i inicialitzo al constructor

Comment (rogermiretin2) [reply src/app/components/barcode-scanner/barcode-scanner.component.ts]: rectifico: hi he posat private originalConsoleError: undefined|((...data: any[]) => void); i ho he deixat a ngOnInit. Alguns tests petaven

File: src/environments/environment.prod.ts [hunk 4]
```diff
@@ -21,5 +21,7 @@ export const environment = {
     cbor: window["env"]["cbor_uri"] || '/api/v2/vp/cbor'
 
   },
-  websocket_uri: window["env"]["websocket_uri"] || '/api/v2/pin'
+  websocket_uri: window["env"]["websocket_uri"] || '/api/v2/pin',
+  LOGS_EMAIL: "domesupport@in2.es",
```
[the remark this answers is not in the document]
Comment (rogermiretin2) [reply src/environments/environment.prod.ts]: fet

**Review by albertorubioin2 — APPROVED** [review]

File: customAssets/i18n/en.json
Comment (rubenmodamioin2) [inline customAssets/i18n/en.json:1]: Debe ir en ingles

File: customAssets/i18n/es.json
Comment (rubenmodamioin2) [inline customAssets/i18n/es.json:1]: Debe ir en castellano

File: customAssets/i18n/en.json
Comment (rubenmodamioin2) [inline customAssets/i18n/en.json:1]: Debe ir en ingles

File: customAssets/i18n/es.json
Comment (rubenmodamioin2) [inline customAssets/i18n/es.json:1]: Debe ir en castellano

**Review by rubenmodamioin2 — CHANGES_REQUESTED** [review]

**Review by rubenmodamioin2 — APPROVED** [review]

**Review by javieralvarezin2 — APPROVED** [review]

**Pull request merged**